operator/pkg/ciliumidentity/reconciler_test.go

```diff
@@ -0,0 +1,721 @@
+			usedInPods:        false,
+			desiredState:      testLbsA,
+			expectedQueueSize: 0, // TODO queue size should be 1 if OP deletes CID
+		},
+		{
+			name:         "cid_in_desired_not_used",
+			cidKey:       cidResourceKey(cidName),
+			usedInPods:   false,
+			desiredState: testLbsA,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			reconciler, queueOps, _, cleanupFunc := testNewReconciler(t, ctx, false)
+			defer cleanupFunc()
+
+			cs := reconciler.clientset.(*k8sClient.FakeClientset)
+			var createCID, updateCID *capi_v2.CiliumIdentity
+			var deleteCIDName string
+			cs.CiliumFakeClientset.PrependReactor("create", "*", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+				pa := action.(k8sTesting.CreateAction)
+				createCID = pa.GetObject().(*capi_v2.CiliumIdentity)
+				return true, nil, nil
+			})
+			cs.CiliumFakeClientset.PrependReactor("update", "*", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+				pa := action.(k8sTesting.UpdateAction)
+				updateCID = pa.GetObject().(*capi_v2.CiliumIdentity)
+				return true, nil, nil
+			})
+			cs.CiliumFakeClientset.PrependReactor("delete", "*", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+				if tc.errorDuringDeletion {
+					return false, nil, fmt.Errorf("forced error on deletion")
+				}
+				pa := action.(k8sTesting.DeleteAction)
+				deleteCIDName = pa.GetName()
+				return true, nil, nil
+			})
+
+			if tc.initialStoreState != nil {
+				if err := reconciler.cidStore.CacheStore().Add(tc.initialStoreState); err != nil {
+					t.Errorf("Error adding CID to cache store: %v", err)
+				}
+			}
+
+			if tc.desiredState != nil {
+				if err := reconciler.upsertDesiredState(tc.cidKey.Name, key.GetCIDKeyFromLabels(tc.desiredState, labels.LabelSourceK8s)); err != nil {
+					t.Errorf("Unexpected error upserting desired state: %v", err)
+				}
+			}
+
+			if tc.usedInPods {
+				reconciler.cidUsageInPods.cidUsageCount[tc.cidKey.Name] = 1
+			}
+
+			if err := reconciler.reconcileCID(tc.cidKey); err != nil {
+				t.Errorf("Unexpected error during reconciliation: %v", err)
+			}
+
+			if !reflect.DeepEqual(createCID, tc.expectedCreate) {
+				t.Errorf("Unexpected createCID result: got %v, want %v", createCID, tc.expectedCreate)
+			}
+			if !reflect.DeepEqual(updateCID, tc.expectedUpdate) {
+				t.Errorf("Unexpected updateCID result: got %v, want %v", updateCID, tc.expectedUpdate)
+			}
+			if deleteCIDName != tc.expectedDelete {
+				t.Errorf("Unexpected deleteCIDName result: got %v, want %v", deleteCIDName, tc.expectedDelete)
+			}
+			if len(queueOps.fakeWorkQueue) != tc.expectedQueueSize {
+				t.Errorf("Unexpected fakeWorkQueue size: got %v, want %v", len(queueOps.fakeWorkQueue), tc.expectedQueueSize)
+			}
+
+			// TODO check for CID deletion too, adapt once we decide how to handle CID deletion
```

Comment:
Could you please add a reference to the Identity GC PR where the discussion is taking place? That will give useful context to a future casual reader.